flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala

```diff
@@ -653,6 +655,36 @@ object FlinkRexUtil {
       rexBuilder,
       converter);
   }
+
+  /** Extracts the default types of operands from a given SqlCall that uses named arguments. */
+  def extractDefaultTypes(call: SqlCall, validator: SqlValidator): Array[RelDataType] = {
+    if (!hasAssignment(call)) {
+      return Array.empty
+    }
+
+    extractDefaultTypes(call.getOperator, validator.getTypeFactory)
+  }
+
+  /** Extracts the default types of operands for a given SqlOperator that uses named arguments. */
+  def extractDefaultTypes(
+      sqlOperator: SqlOperator,
+      typeFactory: RelDataTypeFactory): Array[RelDataType] = {
+    sqlOperator.getOperandTypeChecker
+      .asInstanceOf[SqlOperandMetadata]
```

Comment:
I am a little worried about the design here. If `extractDefaultTypes` is common utils for others, it should handle all kinds of SqlOperator but the current implementation only process Operator that uses named argument. It's not a good design that users should check the implementation because others is not familiar as us.

I suggest we add a class extends SqlCallBinding and move all this into the newly added class. WDYT?